### PR TITLE
API Enhancements

### PR DESCRIPTION
**Pull request by rikner**
Adds new public APIs:

- JNI.GetDirectBufferAddress(buffer:)
- JNI.GetField(fieldName: fieldJavaClassName: from:)
- JNIObject.getField(fieldName: fieldJavaClassName:)

### CODE REVIEW

File: Sources/JNI/JNIFields.swift [hunk 1]
```diff
@@ -25,6 +25,14 @@ public extension JNI {
         try checkAndThrowOnJNIError()
         return try T.fromField(fieldID!, on: javaObject)
     }
+
+    func GetField(_ fieldName: String, fieldJavaClassName: String, from javaObject: JavaObject) throws -> JavaObject {
+        let env = self._env
+        let javaClass = try GetObjectClass(obj: javaObject)
+        let fieldID = env.pointee.pointee.GetFieldID(env, javaClass, fieldName, "L\(fieldJavaClassName);")
```
Comment (ephemer) [inline Sources/JNI/JNIFields.swift:35]: I'm not sure but I think we might need to delete the local ref to the class here after we use it

File: Sources/JNI/JNIObjects.swift [hunk 2]
```diff
@@ -87,6 +87,10 @@ open class JNIObject {
         return try jni.GetField(fieldName, from: self.instance)
     }
 
+    public func getField(_ fieldName: String, fieldJavaClassName: String) throws -> JavaObject {
+        return try jni.GetField(fieldName, fieldJavaClassName: fieldJavaClassName, from: self.instance)
```
Comment (ephemer) [inline Sources/JNI/JNIObjects.swift:91]: Cool looks good, ideally we'd add getters for primitive fields too (Int etc) but we can also do that later when/if we need it

File: Sources/JNI/JNI.swift [hunk 3]
```diff
@@ -64,4 +64,9 @@ public extension JNI {
         let env = self._env
         return env.pointee.pointee.MonitorExit(env, obj)
     }
+
+    func GetDirectBufferAddress(buffer: JavaObject) -> UnsafeMutableRawPointer? {
+        let env = self._env
+        return env.pointee.pointee.GetDirectBufferAddress(env, buffer)
```
Comment (ephemer) [inline Sources/JNI/JNI.swift:70]: 👌

**Review by ephemer — APPROVED** [review]

File: Sources/JNI/JNIFields.swift (hunk 1 above)
Comment (rikner) [reply Sources/JNI/JNIFields.swift:35]: oh yes, totally oversaw this. we do it also in the other methods above :+1

File: Sources/JNI/JNIObjects.swift (hunk 2 above)
Comment (rikner) [reply Sources/JNI/JNIObjects.swift:91]: I think what you mean is already covered by the generic methods above.

**Pull request merged**